walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -37,29 +37,24 @@
                 is larger than meeting request 
         8. return availability list 
         
-        Time Complexity: O(n^2) 
+        Time Complexity: O(N) 
         */
         ArrayList<TimeRange> meetingConflicts = new ArrayList<TimeRange>(); 
         ArrayList<TimeRange> availability = new ArrayList<TimeRange>(); 
 
         //Adds meeting conflicts of required attendees   
-        for (Event getEvent : events) {
-            for (String findAttendee : getEvent.getAttendees()) {
-                if (request.getAttendees().contains(findAttendee)) {
-                    meetingConflicts.add(getEvent.getWhen());
-                } 
+        for (Event getEvent : events) { 
+            if (request.getAttendees().containsAll(getEvent.getAttendees())) { 
+                meetingConflicts.add(getEvent.getWhen());
             }
         }
 
-        //Adds meeting conflicts of optional attendees 
-        for (Event getEvent : events) {
-            for (String optionalAttendee : getEvent.getAttendees()) {
-                if (request.getOptionalAttendees().contains(optionalAttendee) && 
+        //Adds meeting conflicts of optional attendees
+        for (Event getEvent : events) { 
+            if (request.getOptionalAttendees().containsAll(getEvent.getAttendees()) && 
                                                         !(getEvent.getWhen().duration() >= TimeRange.WHOLE_DAY.duration()) &&
                                                         !(getEvent.getWhen().duration() < request.getDuration())) {
```

Comment:
You might want to consider just negating the inequalities to make it easier to understand i.e. instead of !(x >= y) do x<y

Reply:
Thank you for the feedback, I will make sure to delete the negation. That is a very good question, I did sort it using the collection.sort, and I did not take that into consideration when calculating my Time Complexity. I think it would be O(nlogn), since the time complexity of the Collections.sort method is O(nlogn).